x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTests.java

```diff
@@ -347,6 +348,60 @@ public void testRunAsAuthenticationWithDomain() {
         assertThat(test.getDomain(), nullValue());
     }
 
+    public void testCanAccessAcrossDomain() {
+        RealmRef authRealmRef = randomRealmRef(randomBoolean());
```

Comment:
I believe you want `randomRealmRef(true)` here otherwise `getDomain` can be null as commented above. Also we already have tests for this to work with domain. So this new one can focus on when domain is in effect.